appium-dotnet-driver/Appium/AppiumWebElement.cs

```diff
@@ -48,11 +50,111 @@ public abstract class AppiumWebElement : RemoteWebElement,
         public AppiumWebElement(RemoteWebDriver parent, string id)
             : base(parent, id)
         {
+        }
+
+        #region Cache 
+
+        protected Dictionary<string, object> cache = null;
+
+        public virtual void SetCacheValues(Dictionary<string, object> cacheValues)
+        {
+            cache = new Dictionary<string, object>(cacheValues);

```

Comment:
I'd rather use more advanced data structure for the cache, for example LRU cache with limited items size, since the "classic" dictionary may grow up too big if there are many elements in the UI.

Reply:
The cache is per element; so there will only be a handful of attributes in this dictionary